go/pkg/client/bytestream_test.go

```diff
@@ -0,0 +1,294 @@
+)
+
+type logStream struct {
+	logStreamID   string
+	logicalOffset int64
+	finalized     bool
+}
+
+func TestWriteBytesWithOffsetSuccess_LogStream(t *testing.T) {
+	tests := []struct {
+		description string
+		ls          *logStream
+		data        []byte
+		numIterate  int
+		opts        []ByteStreamWriteOption
+		wantBytes   int64
+	}{
+		{
+			description: "valid data with offset 0",
+			ls:          &logStream{logStreamID: "logstream1", logicalOffset: 0},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteSteramOptFinishWrite(false)},
+			numIterate:  3,
+			wantBytes:   int64(len(logStreamData)),
+		},
+		{
+			description: "valid data with non-zero offset",
+			ls:          &logStream{logStreamID: "logstream2", logicalOffset: 4},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(4), ByteSteramOptFinishWrite(false)},
+			numIterate:  3,
+			wantBytes:   int64(len(logStreamData)),
+		},
+		{
+			description: "one big chunk",
+			ls:          &logStream{logStreamID: "logstream3", logicalOffset: 0},
+			data:        logStreamData,
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteSteramOptFinishWrite(true)},
+			numIterate:  1,
+			wantBytes:   int64(len(logStreamData)),
+		},
+		{
+			description: "empty data",
+			ls:          &logStream{logStreamID: "logstream4", logicalOffset: 0},
+			data:        []byte{},
+			opts:        []ByteStreamWriteOption{ByteStreamOptOffset(0), ByteSteramOptFinishWrite(false)},
+			numIterate:  1,
+			wantBytes:   0,
+		},
+	}
+
+	b := setup(t)
+	defer b.shutDown()
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			size := len(test.data)/test.numIterate + 1
+			var start int
+			start = int(test.opts[0].(ByteStreamOptOffset))
+			end := size
+			lsID := test.ls.logStreamID
+			logStreams[lsID] = test.ls
+			ChunkMaxSize(size).Apply(b.client)
+
+			for i := 0; i < test.numIterate; i++ {
+				if end > len(test.data) {
+					end = len(test.data)
+				}
+				if i == test.numIterate-1 {
+					test.opts[1] = ByteSteramOptFinishWrite(true)
+				}
+
+				writtenBytes, err := b.client.WriteBytesWithOptions(b.ctx, lsID, test.data[start:end], test.opts...)
+				if err != nil {
+					t.Errorf("WriteBytesWithOffset() failed unexpectedly: %v", err)
+				}
+				test.opts[0] = test.opts[0].(ByteStreamOptOffset) + ByteStreamOptOffset(writtenBytes)
+				start = end
+				end += size
+			}
+
+			if logStreams[lsID].logicalOffset != test.wantBytes {
+				t.Errorf("WriteBytesWithOffset() = %d, want %d", logStreams[lsID].logicalOffset, test.wantBytes)
+			}
+			if logStreams[lsID].finalized == false {
+				t.Error("WriteBytesWithOffset() didn't correctly finalize logstream")
+			}
+		})
+	}
+}
+
+func TestWriteBytesWithOffsetErrors_LogStream(t *testing.T) {
+	tests := []struct {
+		description string
+		ls          *logStream
+		data        []byte
+		notFound    bool
```

Comment:
I couldn't figure out how this field is used. If you're trying to test for a non-existing stream, I might suggest making the `ls` field of the corresponding test `nil`.

Reply:
Done.